pkg/generators/openapi.go

```diff
@@ -452,26 +433,29 @@ func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
 			extensions = append(extensions, NameValue{parts[0], parts[1]})
 		}
 	}
-	// Next, generate extensions from "tags".
-	for _, tagKey := range sortedMapKeys(tagToExtension) {
-		tagValue, err := getSingleTagsValue(CommentLines, tagKey)
-		if err != nil {
-			return err
-		}
-		if len(tagValue) > 0 {
-			extensions = append(extensions, NameValue{tagToExtension[tagKey], tagValue})
-		}
-	}
-	// If there are generated extensions, write them out to schema.
-	// Example: "x-kubernetes-patch-strategy" : "merge"
-	if len(extensions) == 0 {
+	// Next, generate "tag" extensions (e.g. "+patchMergeStrategy", "+listAttribute=...")
+	tagExtensions := getTagExtensions(CommentLines)
```

Comment:
This is confusing to me. Why do we want two different parsing mechanisms for the tags?

Reply:
I was hoping to address this in a future PR. I was hoping to keep this PR as small and focused as possible. I'll combine the two tags types in this PR if you like.

Reply:
As we agreed offline, I will combine the two parsing mechanisms.

Reply:
I'm still a bit confused by several things...

1. `tagExtensions` vs `extensions` is a confusing name combo. There should either be only one of them, or the name should indicate the reason for there being two.
2. It's still not clear why there are two for loops below and two sources?
3. Now that I have stared at this code a lot, do we have a good reason for not prefixing these new tags with the k8s:openapi-gen prefix?